.github/workflows/pytest.yml

```
@@ -26,3 +26,27 @@ jobs:
       - name: Test with pytest  
         run: |  
           python -m unittest -v tests/test_*.py
+
+  lint:
+    name: lint
+    runs-on: ubuntu-latest
+    needs: [build]
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.9', '3.10', '3.11', '3.12']
+
+    steps:
+      - name: Checkout the source code
+        uses: actions/checkout@v3
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: pip install flake8
+
+      - name: Lint code with flake8
+        run: flake8 .
```

Do we just need flake8, Can we do pylint instead.
```suggestion
      - name: Install dependencies
        run: |
        python -m pip install --upgrade pip
        pip install pylint
      - name: Analysing the code with pylint
         run: |
         pylint $(git ls-files '*.py')
```